app/src/main/java/com/github/se/icebreakrr/ui/map/HeatMap.kt

```diff
@@ -0,0 +1,215 @@
+package com.github.se.icebreakrr.ui.profile
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import com.github.se.icebreakrr.model.location.LocationViewModel
+import com.github.se.icebreakrr.model.profile.ProfilesViewModel
+import com.github.se.icebreakrr.ui.navigation.BottomNavigationMenu
+import com.github.se.icebreakrr.ui.navigation.LIST_TOP_LEVEL_DESTINATIONS
+import com.github.se.icebreakrr.ui.navigation.NavigationActions
+import com.github.se.icebreakrr.ui.navigation.Route
+import com.google.android.gms.maps.model.CameraPosition
+import com.google.android.gms.maps.model.LatLng
+import com.google.firebase.firestore.GeoPoint
+import com.google.maps.android.compose.GoogleMap
+import com.google.maps.android.compose.TileOverlay
+import com.google.maps.android.compose.rememberCameraPositionState
+import com.google.maps.android.heatmaps.Gradient
+import com.google.maps.android.heatmaps.HeatmapTileProvider
+import com.google.maps.android.heatmaps.WeightedLatLng
+
+private val DEFAULT_LOCATION = LatLng(46.5197, 6.6323) // EPFL coordinates
+private const val DEFAULT_ZOOM = 15f
+
+// Define custom gradient colors (from blue to red)
+private val GRADIENT_COLORS =
+    intArrayOf(
+        0x00000000, // Transparent
+        0xFF1FAEF0.toInt(), // IceBreakrrBlue
+        0xFF4B8BE0.toInt(), // Light Blue
+        0xFF7768D0.toInt(), // Purple
+        0xFFA346C0.toInt(), // Pink-purple
+        0xFFCE0E00.toInt() // Red
+        )
+
+private val GRADIENT_START_POINTS =
+    floatArrayOf(
+        0.0f, // Fully transparent at 0% intensity
+        0.2f, // Transition to IceBreakrrBlue
+        0.4f, // Transition to Light Blue
+        0.6f, // Transition to Purple
+        0.8f, // Transition to Pink-purple
+        1.0f // Peak at Red
+        )
+
+private fun generateFakePoints(): List<WeightedLatLng> {
```

Comment:
Precise with a comment the purpose of fake points, so that one that reviews your PR understands easily what they are for (and precise that they are temporary).

Reply:
You are right this is a great advice that will save time for the PR review!